### PR TITLE
[bug] add explanation for resolving block list error

### PR DESCRIPTION
**Pull request by luxe**
A user hit this error message and wasn't sure what to do.  Add more context to the error message.

### CODE REVIEW

**Review by jacobmou — APPROVED** [review]
Thanks for the extra explanation!

File: src/main/java/build/buildfarm/instance/shard/ShardInstance.java [hunk 1]
```diff
@@ -1842,6 +1835,14 @@ public void observe(Operation operation) {
...
   }
 
+  private String blockListErrorMessage() {
+    StringBuilder message = new StringBuilder();
+    message.append("This request is in block list and is forbidden.  ");
+    message.append("To resolve this error, you can tag the rule with 'no-remote'.  ");
+    message.append("You can also adjust the action behavior to attempt a different action hash.  ");
+    return message.toString();
+  }
+
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/ShardInstance.java:1845]: Please present this string in the current location for error messages: https://github.com/bazelbuild/bazel-buildfarm/blob/4f0c6f595126950905e5ea625eacddc06e63cafa/src/main/java/build/buildfarm/instance/server/AbstractServerInstance.java#L145-L200, or something more pertinent for ShardInstance if you feel it is isolated enough to it. StringBuilder is not a facility to be used just for 'long' error messages.

**Review by werkt — CHANGES_REQUESTED** [review]

**Comment by luxe** [comment]
@werkt fixed

**Comment by luxe** [comment]
@werkt bump!

**Comment by werkt** [comment]
This response is very bazel specific. When users come at it with confusion over no-remote, I'll prefer a tool check before advising.

**Review by werkt — APPROVED** [review]

**Comment by luxe** [comment]
> This response is very bazel specific. When users come at it with confusion over no-remote, I'll prefer a tool check before advising.

yes, good idea. It will be cool to see buildfarm work with non-bazel clients.  For now, this helps guide our internal colleagues.

**Pull request merged**